api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -0,0 +1,114 @@
+package org.pmiops.workbench.api;
+
+import org.pmiops.workbench.db.dao.UserRecentResourceService;
+import org.pmiops.workbench.db.dao.WorkspaceService;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.UserRecentResource;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.model.Cohort;
+import org.pmiops.workbench.model.FileDetail;
+import org.pmiops.workbench.model.RecentResource;
+import org.pmiops.workbench.model.RecentResourceResponse;
+import org.pmiops.workbench.firecloud.model.WorkspaceResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.inject.Provider;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@RestController
+public class UserMetricsController implements UserMetricsApiDelegate {
+  Provider<User> userProvider;
+  UserRecentResourceService userRecentResourceService;
+  WorkspaceService workspaceService;
+  FireCloudService fireCloudService;
+  private static final String NOTEBOOKS_WORKSPACE_DIRECTORY = "notebooks/";
+
+  @Autowired
+  UserMetricsController(Provider<User> userProvider,
+      UserRecentResourceService userRecentResourceService,
+      WorkspaceService workspaceService,
+      FireCloudService fireCloudService) {
+    this.userProvider = userProvider;
+    this.userRecentResourceService = userRecentResourceService;
+    this.workspaceService = workspaceService;
+    this.fireCloudService = fireCloudService;
+  }
+
+  /**
+   * Gets the list of all resources recently access by user in order of access date time
+   * @return
+   */
+  @Override
+  public ResponseEntity<RecentResourceResponse> getUserRecentResources() {
+    long userId = userProvider.get().getUserId();
+    RecentResourceResponse recentResponse = new RecentResourceResponse();
+    List<UserRecentResource> userRecentResourceList = userRecentResourceService.findAllResourcesByUser(userId);
+    List<Long> workspaceIdList = userRecentResourceList
+        .stream()
+        .map(UserRecentResource::getWorkspaceId)
+        .distinct()
+        .collect(Collectors.toList());
+
+    Map<Long, String>  workspaceAccessMap = workspaceIdList.stream().collect(Collectors.toMap(id -> id, id -> {
+      Workspace workspace = workspaceService.findByWorkspaceId((long) id);
+      WorkspaceResponse workspaceResponse = fireCloudService
```

Comment:
I don't like that we're making external API calls in this loop. I don't have a better suggestion at the moment, but there should  be a FC API call that gets this information for a batch.

Reply:
After looking at FC's apis, I think the other option is to make a single call to get all workspaces from FC, and then pull out the access levels from the ones you care about. The tradeoff here is that the all-workspaces call can be a large (and slower) response, but it's just one call compared to the possibility of N calls for a single workspace. 

Side note - I don't see it, but FC should have an endpoint to get all permissions the user has across all sorts of things. Right now, they're scattered across different endpoints.

Reply:
I can be persuaded that this is an acceptable option given the current realities - the list workspaces API *is* slow and the FC UI team has also struggled with that. I think what I'd like to see at some point is some pushback on FC to make available a batch permissions endpoint that gives us what we need. I can write that story up and see about getting it prioritized.

Reply:
Sounds good to me - thanks 👍